llvm/include/llvm/IR/DiagnosticInfo.h

```diff
@@ -1092,6 +1093,34 @@ class DiagnosticInfoDontCall : public DiagnosticInfo {
   }
 };
 
+/// Class for emiting warning messages from SYCL pass - PropagateAspectUsage
+class DiagnosticInfoSYCLWarning : public DiagnosticInfo {
```

Comment:
I think name of this diagnostic should be more specific if it is used only for PropagateAspectUsage pass. Like  DiagnosticInfoSYCLUnspecAspect.

Reply:
Personally, I'm fine with the diagnostic class being named in generic way even if it is used in only one place. But, looking at the constructor arguments the diagnostic doesn't seem to be so generic, so we should either rename it to make it less generic or we should change the constructor (and other methods) to be less aspects-oriented to the class is indeed generic.

I'm not sure that we will need such kind of diagnostic any time soon, so renaming as @mlychkov suggests is probably preferable